doc/modules/linear_model.rst

```diff
@@ -110,6 +110,10 @@ its ``coef_`` member::
     >>> reg.intercept_
     0.13636...
 
+The :class:`Ridge` regressor has a classifier variant:
+:class:`RidgeClassifier`. This classifier first converts binary targets to
+``{-1, 1}`` and then treats the problem as a regression task, optimizing
+the same objective as above.
 
```

Comment:
I would add something like:

"""
It might seem questionable to use a (penalized) Least Squares loss to fit
a classification model instead of the more traditional logistic or hinge losses.
However in practice all those models can lead to similar cross-validation scores in terms
of accuracy or area under the ROC curve while the penalized least squares loss
used by the RidgeClassifier allow for a very different choice of the numerical solvers
with distinct computation performance profiles.

This classifier can also be sometimes referred to as a [Least Squares Support Vector Machines](https://en.wikipedia.org/wiki/Least-squares_support-vector_machine) with a linear kernel. 
"""